pkg-manager/plugin-commands-installation/src/recursive.ts

```diff
@@ -257,7 +257,7 @@ export async function recursive (
     if (opts.save !== false) {
       await Promise.all(
         mutatedPkgs
-          .map(async ({ manifest }, index) => writeProjectManifests[index](manifest))
+          .map(async ({ originalManifest, manifest }, index) => writeProjectManifests[index](originalManifest ?? manifest))
```

Comment:
It doesn't seem right to me.

First of all, why do you write the original manifest at all? If it is the original one, then it is already the one that is saved in the file system.

Secondly, this means that the file will not be updated when `opts.save` is `true`. It doesn't make sense. 

Maybe what you need to do is changing line 257 to:

```
if (opts.save !== false && !opts.lockfileOnly) {
```

Reply:
My proposed fix may well be wrong, but I'll add some context around how I got here.

Elsewhere in the repo I noticed `.manifest` and `.originalManifest` are both modified to apply changes like removing dependencies and _something_ in `@pnpm/resolve-dependencies` that I assume handles adding and upgrading.

* Removing, replaces manifest object properties.
   https://github.com/pnpm/pnpm/blob/54128c92a27cb35d6ce3d39ff339ba2161c954d3/pkg-manager/core/src/install/index.ts#L749-L752
* Updating and adding (I think), mutates input manifest objects.
   https://github.com/pnpm/pnpm/blob/54128c92a27cb35d6ce3d39ff339ba2161c954d3/pkg-manager/resolve-dependencies/src/updateProjectManifest.ts#L45-L57

Interpretation as such was "`originalManifest` must mean the manifest without mutations from the `readPackage` hook".

Reply:
The original manifest is modified, sure. If I remember correctly, original manifest is only not modified by changes done by the hooks.

Reply:
actually, you might be right. This is what you had to change. The names of variables are confusing.